rclcpp/include/rclcpp/strategies/static_memory_strategy.hpp

```diff
@@ -27,40 +27,64 @@ namespace memory_strategies
 
 namespace static_memory_strategy
 {
-  struct ObjectPoolBounds
-  {
-  public:
-    size_t max_subscriptions_;
-    size_t max_services_;
-    size_t max_clients_;
-    size_t max_executables_;
-    size_t pool_size_;
-
-    ObjectPoolBounds(size_t subs = 10, size_t services = 10, size_t clients = 10,
-      size_t executables = 1, size_t pool = 1024)
-    : max_subscriptions_(subs), max_services_(services), max_clients_(clients), max_executables_(
-        executables), pool_size_(pool)
-    {}
-  };
+struct ObjectPoolBounds
+{
+public:
+  size_t max_subscriptions_;
+  size_t max_services_;
+  size_t max_clients_;
+  size_t max_executables_;
+  size_t max_guard_conditions_;
+  size_t pool_size_;
```

Comment:
Public members shouldn't have a trailing `_`.

I think @dirk-thomas is right, this class should be just like ROS messages where the attributes are public and there are setters for method chaining if desired and the constructor takes no arguments but sets reasonable defaults.